src/gui/qgsdatasourcemanagerdialog.cpp

```diff
@@ -90,7 +110,8 @@ QgsDataSourceManagerDialog::~QgsDataSourceManagerDialog()
 
 void QgsDataSourceManagerDialog::openPage( const QString &pageName )
 {
-  const int pageIdx = mPageNames.indexOf( pageName );
+  // TODO -- this is actually using provider keys, not provider names!
+  const int pageIdx = mPageProviderKeys.indexOf( pageName );
```

Comment:
@nyalldawson, the "Vector" page and the "GeoPackage" page have the same "ogr" provider key, so it is not possible to open the GeoPackage page using the openPage function, isn't it?

Reply:
Actually the "key" referred to here is the QgsSourceSelectProvider providerKey -- it doesn't have to correspond to a QgsDataProvider provider key. So the Geopackage one could (and should) be changed to "gpkg" or similar.

Reply:
> So the Geopackage one could (and should) be changed to "gpkg"

@nyalldawson, I agree. Currently, it seems not possible to add the "Add GeoPackage layer" item to the Layer->Add layer menu: see #51534.